schema/object.go

```diff
@@ -363,7 +364,7 @@ func (o *ObjectSchema) validateStruct(data any) error {
 func (o *ObjectSchema) validateSchemaCompatibility(schemaType Object) error {
 	fieldData := map[string]any{}
 	// Validate IDs. This is important because the IDs should match.
-	if !schemaType.HasLooseIDMatching() && schemaType.ID() != o.ID() {
+	if !schemaType.HasLooseIDMatching() && !o.HasLooseIDMatching() && schemaType.ID() != o.ID() {
```

Comment:
OK; so that's interesting. You're matching IDs only if *neither* object has "loose" matching. If you ever have it set on only one of them ... is that OK, should it be diagnosed, should you *check* the ID because the two don't agree that it *shouldn't* be checked?? And it might not be a bad idea to capture this in the comment.

Reply:
I added some extra language to the interface regarding that. One-way enforcement would cause far more confusion, and I don't think it would help in any scenarios. The current behavior is intentional.